### ISSUE TITLE
Files written with MMCIFIO/PDBIO cannot be use with SeqIO 

### ISSUE THREAD

**Issue by LivC182** [biopython/biopython#3490]
### Setup
```
import sys; print(sys.version)
3.6.10 |Anaconda, Inc.| (default, Jan  7 2020, 21:14:29) [GCC 7.3.0]
import platform; print(platform.python_implementation()); print(platform.platform())
CPython
Linux-5.4.0-62-generic-x86_64-with-debian-buster-sid
import Bio; print(Bio.__version__)
1.73
```
### Expected behaviour
After writing a file with ```MMCIFIO/PDBIO``` i want to fetch the seqence of the written file using ```SeqIO```. When I then parse the file using SeqIO, I expect to get the sequences for all the chains in a string format.

### Actual behaviour

However, this does not work. I assume this is because the footer is not written for CIF files (probably same reason for PDB files)

### Steps to reproduce
`parser = MMCIFParser() #or PDB parser`
`structure = parser.get_structure('x',path) #path = location of full structure as downloaded from rcsb.org`
`io= MMCIFIO() # or io=PDBIO()`
`io.set_structure(structure)`
`io.save(selected_chain_file, SelectChain('B')) #selected_chain_file = place to save structure on memory`
`query_seqres = SeqIO.parse('selected_chain_file', 'cif-seqres')`
`print([chain.seq for chain in query_seqres])`
```
class SelectChain(Select):
    def __init__(self, chain_id):
        self.chain_id = chain_id
    def accept_chain(self, chain):
        if chain.id == self.chain_id:
            return 1
        else:
            return 0  
```

**Comment by peterjc** [comment]
Do you get an exception error message?

**Comment by LivC182** [comment]
@peterjc, no. It just enters the for loop and then exits, because there are no chains to iterate over.

**Comment by JoaoRodrigues** [comment]
Do you have a representative structure we could use to reproduce this issue?

**Comment by LivC182** [comment]
@JoaoRodrigues any pdb/cif you want. Try for 1A1B (any protein chain)

**Comment by LivC182** [comment]
@JoaoRodrigues also another issue that I just identified when saving structures for example for PDB ID: 5AYY (cif format) if I select chain A to be written for some reason it also writes chain J as chain B (NMT hetatm) and chain S as chain C (water). 

Same behaviour is observed if I use 5AYY in the PDB Format

**Comment by JoaoRodrigues** [comment]
Thanks. These seem to be 3 separate issues:

- mmCIF to sequence results in an empty sequence
- Selection/Writing writes more than intended.
- Xs in sequence records. This last one is normal. It's not a random char, it's probably that the residue code maps to a residue name that matches Uracyl. The Xs are also normal, it's the standard character for non-protein residues. What would be the expected behavior for you here?

**Comment by LivC182** [comment]
@JoaoRodrigues  you are right concerning the last point. It is mostly an issue that only comes due to how `MMCIFIO()` writes the structure. So once the Selection/Writing is fixed this should not be an issue anymore. People who want to write protein chains can ask for seqs and people who write PDBBlocks for molecules will use RdKit to generate smiles out of them.

Concerning the first point you can get the seq present using `seq-atom` instead of `seq-seqres`. However, this has a serious downside: some protein structures have breaks within the chain, mostly due to disordered regions, hence you cannot retrieve missing residues from  `seq-atom`

Regarding Selection/Writing I expect just one chain to be written. I know the PDB/mmCIF files are very messy: ligands who are not covalently attacked to a protein having the same chain as the protein (or ligands who are covalently attached to a protein and have a different chain), broken ligands given different chains, and the list can continue. However, if I want a specific chain to be written I except just that chain.

**Comment by LivC182** [comment]
@JoaoRodrigues I have done some more digging and I think the issue with writing files for a specific chain comes from the fact that your script takes the chain name from `_atom_site.auth_asym_id` column instead of `_atom_site.label_asym_id` (on cif files) column.

**Comment by LivC182** [comment]
@JoaoRodrigues another issue with SeqIO happens for PDB ID: 4B74
```
query_seqres = SeqIO.parse('path_to|_file.cif', 'cif-atom')
chains = [str(chain.seq) for chain in query_seqres]
print(chains)
```
Output:
`['GSVVIVGRIILSGSGS', 'GSVVIVGRIILSXXXS']`
Expected output:
`['MGSSHHHHHHSSGLVPRGSHMGSVVIVGRIILSGSGSITAYSQQTRGLLGCIITSLTGRDKNQVEGEVQVVSTATQSFLATCVNGVCWTVYHGAGSKTLAGPKGPITQMYTNVDQDLVGWQAPPGARSLTPCTCGSSDLYLVTRHADVIPVRRRGDSRGSLLSPRPVSYLKGSSGGPLLCPSGHAVGIFRAAVCTRGVAKAVDFVPVESMETTMRSPVFTDNSSPPAVPQSFQVAHLHAPTGSGKSTKVPAAYAAQGYKVLVLNPSVAATLGFGAYMSKAHGIDPNIRTGVRTITTGAPVTYSTYGKFLADGGCSGGAYDIIICDECHSTDSTTILGIGTVLDQAETAGARLVVLATATPPGSVTVPHPNIEEVALSNTGEIPFYGKAIPIEAIRGGRHLIFCHSKKKCDELAAKLSGLGINAVAYYRGLDVSVIPTIGDVVVVATDALMTGYTGDFDSVIDCNTCVTQTVDFSLDPTFTIETTTVPQDAVSRSQRRGRTGRGRRGIYRFVTPGERPSGMFDSSVLCECYDAGCAWYELTPAETSVRLRAYLNTPGLPVCQDHLEFWESVFTGLTHIDAHFLSQTKQAGDNFPYLVAYQATVCARAQAPPPSWDQMWKCLIRLKPTLHGPTPLLYRLGAVQNEVTLTHPITKYIMACMSADLEVVT', 'MGSSHHHHHHSSGLVPRGSHMGSVVIVGRIILSGSGSITAYSQQTRGLLGCIITSLTGRDKNQVEGEVQVVSTATQSFLATCVNGVCWTVYHGAGSKTLAGPKGPITQMYTNVDQDLVGWQAPPGARSLTPCTCGSSDLYLVTRHADVIPVRRRGDSRGSLLSPRPVSYLKGSSGGPLLCPSGHAVGIFRAAVCTRGVAKAVDFVPVESMETTMRSPVFTDNSSPPAVPQSFQVAHLHAPTGSGKSTKVPAAYAAQGYKVLVLNPSVAATLGFGAYMSKAHGIDPNIRTGVRTITTGAPVTYSTYGKFLADGGCSGGAYDIIICDECHSTDSTTILGIGTVLDQAETAGARLVVLATATPPGSVTVPHPNIEEVALSNTGEIPFYGKAIPIEAIRGGRHLIFCHSKKKCDELAAKLSGLGINAVAYYRGLDVSVIPTIGDVVVVATDALMTGYTGDFDSVIDCNTCVTQTVDFSLDPTFTIETTTVPQDAVSRSQRRGRTGRGRRGIYRFVTPGERPSGMFDSSVLCECYDAGCAWYELTPAETSVRLRAYLNTPGLPVCQDHLEFWESVFTGLTHIDAHFLSQTKQAGDNFPYLVAYQATVCARAQAPPPSWDQMWKCLIRLKPTLHGPTPLLYRLGAVQNEVTLTHPITKYIMACMSADLEVVT']`

**Comment by JoaoRodrigues** [comment]
I'll have a look at this this weekend then.

**Comment by JoaoRodrigues** [comment]
We should break this into separate issues, but the main reason behind the first issue (empty sequence after selecting/saving) is because `SeqIO.PdbIO.CifSeqresIterator` reads from `_entity_poly_seq` entries, which our writers do not write. The docstring of the class explicitly mentions:

>The sequences are derived from the _entity_poly_seq entries in the mmCIF file, not the atoms of the 3D structure.

Same for the `PdbSeqresIterator`.

I'll check how hard it'll be to add these records to MMCIFIO/PDBIO.

**Comment by JoaoRodrigues** [comment]
As for your second issue:

>@JoaoRodrigues I have done some more digging and I think the issue with writing files for a specific chain comes from the fact that your script takes the chain name from _atom_site.auth_asym_id column instead of _atom_site.label_asym_id (on cif files) column.
Also looks like even the structure perser MMCIFParser() takes the chain from _atom_site.auth_asym_id column instead of _atom_site.label_asym_id

This is a conscious choice of ours to maintain backwards compatibility with the PDB format. PDB files have one chain id field, whereas mmCIF files have two: one attributed by the authors of the strucure (`_atom_site.auth_asym_id`) and one automatically assigned by software (`_atom_site.label_asym_id`). There is a similar convention for other fields in the files. We decided to assign our chain IDs based on the `auth` field since that matches the old PDB format. Otherwise we'd run into issues where you'd do 'model["A"]` for a PDB and CIF of the same structure and return different results.

That is why when you select on chain A, you get "multiple" chains. If you open the mmCIF file, you'll see that the second to last field is always A, and that's the `auth_asym_id`, e.g.:
```
HETATM 2120 O O4  . NTM B ? .   ? 107.841 40.347 10.205  1.0  72.71  401 A 1
HETATM 2121 O O   . HOH C ? .   ? 119.796 61.521 17.515  1.0  29.96  501 A 1
```

Does this make sense to you?

**Comment by JoaoRodrigues** [comment]
As for your last issue, on 4b74 (please open multiple issues in situations like this!). The issue here is that the PDB is out of order and indeed, the parser throws a warning:

```python

In [1]: from Bio import SeqIO

In [2]: q = SeqIO.parse("./4b74.cif", "cif-atom")

In [3]: chains = [str(chain.seq) for chain in q]
/Users/jrodrigu/foss-work/biopython/Bio/PDB/StructureBuilder.py:89: PDBConstructionWarning: WARNING: Chain A is discontinuous at line 9602.
  warnings.warn(
/Users/jrodrigu/foss-work/biopython/Bio/PDB/StructureBuilder.py:89: PDBConstructionWarning: WARNING: Chain B is discontinuous at line 9659.
  warnings.warn(
/Users/jrodrigu/foss-work/biopython/Bio/PDB/StructureBuilder.py:89: PDBConstructionWarning: WARNING: Chain A is discontinuous at line 9716.
  warnings.warn(
/Users/jrodrigu/foss-work/biopython/Bio/PDB/StructureBuilder.py:89: PDBConstructionWarning: WARNING: Chain B is discontinuous at line 10075.
  warnings.warn(
/Users/jrodrigu/foss-work/biopython/Bio/SeqIO/PdbIO.py:82: BiopythonParserWarning: Ignoring out-of-order residues after a gap
  warnings.warn(

In [4]: chains
Out[4]: ['GSVVIVGRIILSGSGS', 'GSVVIVGRIILSXXXS']
```

Not ideal, but this is an issue of the structure rather than the parser.

**Comment by LivC182** [comment]
@JoaoRodrigues thank you for your fast answer, and apologies for putting all the issues in a single entry. I thought it makes more sense to put everything here since the only way I actually found this is because I used all the functions together. 

Everything makes sense now, and thank you for clearing everything up. I think that by adding the header for a specific chain when using` MMCIFIO/PDBIO `would be very helpful since some entries might be missing residues in the 3D structure so `cif-atom` would miss them as well. Another work-around to get the sequence would be to iterate over the chain by using `Selection.unfold_entities(structure, 'chain')` and simply iterate over the residues and add them to a string if they are protein atoms.

**Comment by LivC182** [comment]
@JoaoRodrigues actually I doubled checked, to make sure I fully understand, how the `SeqIO.parse()` is implemented for cif files and and cif-seqres:
https://github.com/biopython/biopython/blob/bdcaa7312217229dde48b206b31e5c2f06fc3613/Bio/SeqIO/PdbIO.py#L395
It actually states that the chain is taken from:
`_pdbx_poly_seq_scheme.asym_id` instead of  `_pdbx_poly_seq_scheme.pdb_strand_id `. Usually is not an issue since protein chains are numbered from A but in the case of : `6qlq.cif and 1a0t.cif` they actually start from B and P in their respective PDB files.

**Comment by JoaoRodrigues** [comment]
Then we should address this so that the SeqIO and PDB modules agree.
